packages/sol-doc/src/solidity_doc_generator.ts

```diff
@@ -0,0 +1,306 @@
+            if (!_.isUndefined(devdocMethodIfExists)) {
+                const devdocMethodDetailsIfExist = devdocMethodIfExists.details;
+                if (!_.isUndefined(devdocMethodDetailsIfExist)) {
+                    details = devdocMethodDetailsIfExist;
+                }
+            }
+        }
+    }
+    return details;
+}
+
+function _genMethodDoc(
+    abiDefinition: MethodAbi | FallbackAbi,
+    devdocIfExists: DevdocOutput | undefined,
+): SolidityMethod {
+    const name = abiDefinition.type === 'fallback' ? '' : abiDefinition.name;
+
+    const { parameters, methodSignature } =
+        abiDefinition.type === 'fallback'
+            ? { parameters: [], methodSignature: `${name}()` }
+            : _genMethodParamsDoc(name, abiDefinition.inputs, devdocIfExists);
+
+    const comment = _devdocMethodDetailsIfExist(methodSignature, devdocIfExists);
+
+    const returnType =
+        abiDefinition.type === 'fallback'
+            ? { name: '', typeDocType: TypeDocTypes.Intrinsic }
+            : _genMethodReturnTypeDoc(abiDefinition.outputs, methodSignature, devdocIfExists);
+
+    const returnComment =
+        _.isUndefined(devdocIfExists) || _.isUndefined(devdocIfExists.methods[methodSignature])
+            ? undefined
+            : devdocIfExists.methods[methodSignature].return;
+
+    const isConstant = abiDefinition.type === 'fallback' ? true : abiDefinition.constant;
+
+    const methodDoc: SolidityMethod = {
+        isConstructor: false,
+        name,
+        callPath: '',
+        parameters,
+        returnType,
+        returnComment,
+        isConstant,
+        isPayable: abiDefinition.payable,
+        comment,
+    };
+    return methodDoc;
+}
+
+function _genEventDoc(abiDefinition: EventAbi): Event {
+    const eventDoc: Event = {
+        name: abiDefinition.name,
+        eventArgs: _genEventArgsDoc(abiDefinition.inputs, undefined),
+    };
+    return eventDoc;
+}
+
+function _genEventArgsDoc(args: EventParameter[], devdocIfExists: DevdocOutput | undefined): EventArg[] {
+    const eventArgsDoc: EventArg[] = [];
+
+    for (const arg of args) {
+        const name = arg.name;
+
+        const type: Type = {
+            name: arg.type,
+            typeDocType: TypeDocTypes.Intrinsic,
+        };
+
+        const eventArgDoc: EventArg = {
+            isIndexed: arg.indexed,
+            name,
+            type,
+        };
+
+        eventArgsDoc.push(eventArgDoc);
+    }
+    return eventArgsDoc;
+}
+
+/**
+ * Extract documentation for each method paramater from @param params.
+ */
+function _genMethodParamsDoc(
+    name: string,
+    abiParams: DataItem[],
+    devdocIfExists: DevdocOutput | undefined,
+): { parameters: Parameter[]; methodSignature: string } {
+    const parameters: Parameter[] = [];
+    for (const abiParam of abiParams) {
+        const parameter: Parameter = {
+            name: abiParam.name,
+            comment: '',
```

Comment:
No parameter comments?

Reply:
It's a 2-stage buildup in this method here.  First stage is data from ABI, second is from devdoc.  In the ABI stage, we don't have the information for the parameter comment, but we must declare the field here because it's required on the `Parameter` type.  Later in this method, we set this field to something meaningful, iff we find relevant text in the devdoc.